llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -937,6 +949,44 @@ Value *InstrLowerer::getCounterAddress(InstrProfCntrInstBase *I) {
   return Builder.CreateIntToPtr(Add, Addr->getType());
 }
 
+Function *InstrLowerer::createRMWOrFunc() {
```

Comment:
Can you document this function with basic C code to make is easy to understand what it's doing.